64_FFT/app_resources/shader.comp.hlsl

```diff
@@ -32,17 +31,16 @@ struct SharedMemoryAccessor {
 
 };
 
-struct Accessor {
+struct Accessor
+{
 	void set(uint32_t idx, nbl::hlsl::complex_t<scalar_t> value) 
 	{
-		vk::RawBufferStore< vector<scalar_t, 2> >(pushConstants.outputAddress + sizeof(vector<scalar_t, 2>) * idx, vector<scalar_t, 2>(value.real(), value.imag()));
+		vk::RawBufferStore< nbl::hlsl::complex_t<scalar_t> >(pushConstants.outputAddress + sizeof(nbl::hlsl::complex_t<scalar_t>) * idx, value);
 	}
 	
 	void get(uint32_t idx, NBL_REF_ARG(nbl::hlsl::complex_t<scalar_t>) value) 
 	{
-		vector<scalar_t, 2> aux = vk::RawBufferLoad< vector<scalar_t, 2> >(pushConstants.inputAddress + sizeof(vector<scalar_t, 2>) * idx);
-		value.real(aux.x);
-		value.imag(aux.y);
+		value = vk::RawBufferLoad< nbl::hlsl::complex_t<scalar_t> >(pushConstants.inputAddress + sizeof(nbl::hlsl::complex_t<scalar_t>) * idx);
```

Comment:
what's wrong with using the BDA accessor?

Reply:
ah ok you need separate input and output address... take the DoublePtrAccessor from example 10 counting sort rename it to something better and stick it in Nabla common headers.

Reply:
Problem was SFINAE resolution (atomics weren't templated so you could only instantiate BDA accessors with integral types). If it's been changed I'll switch to that otherwise I'll see about fixing it myself the way you did references with dummy template arguments